### PR TITLE
pythonPackages.pytestCheckHook: Add disabledTestFiles option

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by jonringer — COMMENTED** [review]
diff LGTM
but should probably target `staging` as this is a commonly used testing hook

**Review by jonringer — APPROVED** [review]

**Comment by jonringer** [comment]
@GrahamcOfBorg build python3Packages.pandas

**Comment by SuperSandro2000** [comment]
ofborg does not want to give me a log...

**Comment by jonringer** [comment]
@GrahamcOfBorg build python3Packages.pandas

**Review by FRidh — APPROVED** [review]

**Comment by mweinelt** [comment]
Hey @jonringer @FRidh! As you have both approved of this change, what's keeping it from being merged? Python batch upgrades? If so we should add it to the project.

**Review by mweinelt — APPROVED** [review]

**Comment by jonringer** [comment]
I just wanted to verify that there wasn't some edge case which broke some builds, if the pandas build wasn't a false negative, I would have merged a while ago

**Comment by jonringer** [comment]
was able to build azure-cli locally, pandas got to scipy and i didn't want to finish the build + test

**Comment by dotlambda** [comment]
This doesn't seem to work with more than one file.

File: pkgs/development/interpreters/python/hooks/pytest-check-hook.sh [hunk 1]
```diff
@@ -36,6 +37,13 @@ function pytestCheckPhase() {
         disabledTestsString=$(_pytestComputeDisabledTestsString "${disabledTests[@]}")
       args+=" -k \""$disabledTestsString"\""
     fi
+    for file in "${disabledTestFiles[@]}"; do
+      if [ ! -f "$file" ]; then
+        echo "Disabled test file \"$file\" does not exist. Aborting"
+        exit 1
+      fi
+      args+=" --ignore=$file"
```
Comment (dotlambda) [inline pkgs/development/interpreters/python/hooks/pytest-check-hook.sh:45]: This seems problematic if the filename contains whitespace.

**Comment by mweinelt** [comment]
> This doesn't seem to work with more than one file.

How does it fail?

**Comment by dotlambda** [comment]
> How does it fail?

E.g.
```
Executing pytestCheckPhase
Disabled test file "tests/unit/create/test_creator.py tests/unit/seed/embed/test_bootstrap_link_via_app_data.py" does not exist. Aborting
```

**Pull request merged**